external/vcm/vcm/cubedsphere/regridz.py

```diff
@@ -155,25 +171,38 @@ def _regrid_given_delp(
         delp_coarse, delp_fine, x_dim=x_dim, y_dim=y_dim
     )
     phalf_coarse_on_fine = pressure_at_interface(
-        delp_coarse_on_fine, dim_center=z_dim, dim_outer=RESTART_Z_OUTER
+        delp_coarse_on_fine, dim_center=z_dim, dim_outer=z_dim_outer
     )
     phalf_fine = pressure_at_interface(
-        delp_fine, dim_center=z_dim, dim_outer=RESTART_Z_OUTER
+        delp_fine, dim_center=z_dim, dim_outer=z_dim_outer
     )
+    pfull_fine = pressure_at_midpoint_log(delp_fine, dim=z_dim)
+    pfull_coarse_on_fine = pressure_at_midpoint_log(delp_coarse_on_fine, dim=z_dim)
 
     ds_regrid = xr.zeros_like(ds)
     for var in ds:
-        ds_regrid[var] = regrid_vertical(
+        remapped = regrid_vertical(
```

Comment:
It looks like `regrid_vertical` already has an input `iv` which supposedly indicates what kind of variables are being regridded. Though here `iv` doesn't appear to differ for different variables. I was going to suggest wrapping the boolean `temperature` in `regrid_vertical` so you don't have to have another if statement below. But then it needs the extra arguments, which is not ideal. So it's probably fine as it is, though I'm curious if `iv` should've been different?

Reply:
That's right -- we've always used `iv = 1` in our coarsening work.  It's been a long time since we thought about this, and I think @oliverwm1 might have been the one to initially select that. It's possible we decided to just use the generic remapping option (`iv = 1`) to keep things simple and predictable.

Reply:
The `iv = 1` in conjunction with `kord = 1`, the original monotone PPM constraint, should be fine for your purposes; this will not create new extrema that need an additional limiter to enforce positivity, and doesn't apply any special boundary conditions at the top and the bottom of the source domain.